src/components/CodeInput/PCodeInput.vue

```diff
@@ -63,10 +66,28 @@
 
   const textarea = ref()
   const { source, target } = useScrollLinking()
+  const viewWidth = ref(0)
 
-  const valueLines = computed(() => internalValue.value.split(/\r|\r\n|\n/))
-  const lines = computed(() => Math.max(valueLines.value.length, 1))
-  const rows = computed(() => valueLines.value.length)
+  const lineSplitRegex = /\r|\r\n|\n/
+  const valueLines = computed(() => {
+    if (internalValue.value !== '') {
+      return internalValue.value.split(lineSplitRegex)
+    }
+
+    if (props.placeholder) {
+      return props.placeholder.split(lineSplitRegex)
+    }
+
+    return []
+  })
+  const lines = computed(() => Math.max(valueLines.value.length, props.minLines ?? 1))
+  const lineHeight = computed(() => {
+    if (textarea.value) {
+      return parseFloat(getComputedStyle(textarea.value).lineHeight)
+    }
+
+    return 0
+  })
```

Comment:
There's a composition in vue-compositions for this called [useComputedStyle](https://github.com/prefecthq/vue-compositions/tree/main/src/useComputedStyle).